Implement vaccine certificate

thanghq-zinza opened the pull request.

thanghq-zinza committed:
fix: Separate auth pages and main pages

thanghq-zinza committed:
feat: Add Vaccine certificate page

thanghq-zinza committed:
fix:  Fix Vaccine certification page

Fix empty injection record
Fix certification card color

thanghq-zinza committed:
feat: Add Vaccine registration lookup page

thanghq-zinza committed:
feat: implement Vaccine certificate

Add Vaccine certificate page
Add Vaccine Registration Lookup page
Add Account edit page

thanghq-zinza committed:
fix: Fix useQuery key for 'ward' query

hungnp194 commented on src/app/(mainPage)/layout.tsx:
```diff
@@ -0,0 +1,12 @@
+
+const MainPageLayout: FC<PropsWithChildren> = ({ children }) => {
+  return (
+    <>
+      <ClientSideLayout>{children}</ClientSideLayout>
+    </>
```
```suggestion
    <ClientSideLayout>{children}</ClientSideLayout>
```

Có 1 element thì dùng Fragment chi?

hungnp194 commented on src/app/(mainPage)/portal/(accountPage)/vaccine-certificate/page.tsx:
```diff
@@ -0,0 +1,42 @@
+}
+
+const fetchData: InjectionRecord[] = [
+  // {
+  //   injectionTime: '08/09/2021 - 16:56',
+  //   vaccineType: '	COVID-19 Vaccine AstraZeneca',
+  //   batchNumber: 'NJ0342',
+  //   vaccinationPoint: 'TYT Dịch Vọng Hậu'
+  // },
+  // {
+  //   injectionTime: '08/02/2022 - 04:23',
+  //   vaccineType: '	COVID-19 Vaccine AstraZeneca',
+  //   batchNumber: 'NJ0343',
+  //   vaccinationPoint: 'TYT Dịch Vọng Hậu'
+  // }
```
code ko dùng thì xóa đi, ko comment

hungnp194 commented on src/app/(mainPage)/portal/(accountPage)/vaccine-certificate/page.tsx:
```diff
@@ -0,0 +1,42 @@
+import { Grid } from '@mui/material';
+import CertificateCard from '@src/components/portal/vaccine-certificate/CertificateCard';
+import CertificateInfo from '@src/components/portal/vaccine-certificate/CertificateInfo';
+import React, { FC } from 'react';
+export interface InjectionRecord {
+  injectionTime: string;
```
phần import cách 1 dòng ra với code chính

hungnp194 commented on src/components/portal/PortalNavigation.tsx:
```diff
@@ -0,0 +1,73 @@
+
+  return (
+    <CustomTabs value={tab}>
+      {tabs.map((tab, index) => (
+        <CustomTab
+          key={index}
+          label={tab.label}
+          onClick={() => handleClick(tab.path)}
+        />
+      ))}
```
Dùng Link component, ko dùng onClick
Dùng onClick sẽ mất đi nhiều tính năng hữu dụng của thẻ a. Ví dụ như context menu, mouse over, mouse wheel

hungnp194 commented on src/components/portal/lookup-injection-registration/VaccinationRegistrationTable.tsx:
```diff
@@ -0,0 +1,120 @@
+};
+
+interface VaccinationRegistrationTableProps {
+  vaccineRegistrationList: VaccinationRegistration[];
```
```suggestion
  vaccineRegistrations: VaccinationRegistration[];
```

chú ý cách đặt tên, ưu tiên số nhiều số ít, hạn chế dùng List

hungnp194 commented on src/components/portal/lookup-injection-registration/VaccinationRegistrationTable.tsx:
```diff
@@ -0,0 +1,120 @@
+          ))}
+        </TableBody>
+      </Table>
+      {vaccineRegistrationList.length === 0 && (
```
```suggestion
      {!vaccineRegistrationList.length && (
```
Viết ngắn gọn thôi

hungnp194 commented on src/components/portal/vaccine-certificate/CertificateCard.tsx:
```diff
@@ -0,0 +1,77 @@
+}
+
+const CertificateCard: FC<CertificateCardProps> = ({ injectionRecordList }) => {
+  const getCertificationCardColor = () => {
+    if (injectionRecordList.length === 0) return red[700];
+    if (injectionRecordList.length === 1) return orange[600];
+    if (injectionRecordList.length >= 1) return green[500];
+  };
+  return (
```
các block code độc lập như thế này thì cách nhau 1 dòng

thanghq-zinza committed:
feat: Add View account and Logout button in Header

thanghq-zinza committed:
fix: Fix bugs

Change Portal nav tab to Link component.
Separate between logic code blocks.
Change plural variable from _List to _s.
Clear unuse code.

thanghq-zinza committed:
fix: use useCallback for getPath in PortalNav

thanghq-zinza committed:
fix: Clear unsuse code
